app/uk/gov/hmrc/gform/fileupload/FileUploadService.scala

```diff
@@ -61,7 +58,7 @@ class FileUploadService(
     val fileNamePrefix = s"${submissionRef.withoutHyphens}-$date"
     val reconciliationId = ReconciliationId.create(submissionRef)
     val metadataXml = MetadataXml.xmlDec + "\n" + MetadataXml
-      .getXml(submission, reconciliationId, summaries.pdfSummary, dmsSubmission, numberOfAttachments)
+      .getXml(submission, reconciliationId, summaries.pdfSummary, dmsSubmission, submission.noOfAttachments)
```

Comment:
Since we're passing 'submission' here, do we need to pass 'submission.noOfAttachments' too?

Reply:
Good catch